addons/pos_loyalty/static/src/js/ProductScreen.js

```diff
@@ -105,4 +105,8 @@ patch(ProductScreen.prototype, "pos_loyalty.ProductScreen", {
             selectedLine.order._updateRewards();
         }
     },
+    async _barcodeProductAction(code) {
+        this._super(code);
```

Comment:
await the super call
```suggestion
        await this._super(code);
```